### PR TITLE
Implement entity move to list

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/Entities/EntityManager.cpp [hunk 1]
```diff
@@ -136,13 +137,27 @@ namespace OpenLoco::EntityManager
         else
         {
             auto* previousEntity = get<EntityBase>(previousId);
-            previousEntity->next_thing_id = nextId;
+            if (previousEntity == nullptr)
+            {
+                std::cerr << "Invalid previous entity id. Entity linked list corrupted?";
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Entities/EntityManager.cpp:142]: I think it would be best to stick with `Console::error` for this.

Comment (duncanspumpkin) [reply src/OpenLoco/Entities/EntityManager.cpp:142]: Argh I couldn't find what we were using for errors. I could only remember log_error from OpenRCT2.

File: src/OpenLoco/Entities/EntityManager.h [hunk 2]
```diff
@@ -12,12 +12,13 @@ namespace OpenLoco::Vehicles
...
 namespace OpenLoco::EntityManager
 {
-    constexpr size_t numEntityLists = 6;
+    constexpr size_t numEntityLists = 7;
     constexpr size_t maxEntities = 20000;
 
     enum class EntityListType
     {
         null,
+        null2,
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Entities/EntityManager.h:21]: Could you add comment denoting which of these lists is used for what entities?

Comment (duncanspumpkin) [reply src/OpenLoco/Entities/EntityManager.h:21]: yeah at the time i didn't know but i can now say its the null money entities

Comment (AaronVanGeffen) [reply src/OpenLoco/Entities/EntityManager.h:21]: Yeah, I asked because I'd seen something to that end on Discord. :) Thanks

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**